service/matching/taskListManager.go

```diff
@@ -323,6 +409,16 @@ func (c *taskListManagerImpl) getTask(ctx context.Context) (*getTaskResult, erro
 		}()
 	}
 
+	// Wait till long poll expiration for token. If token acquired, proceed.
+	stopWatch := c.metricsClient.StartTimer(scope, metrics.PollThrottleLatency)
+	err := c.rateLimiter.Wait(ctx, c.config.LongPollExpirationInterval)
```

Comment:
I've just realized that this approach is not going to work for the following scenario:
1. Rate limit of 10 polls per second
2. No tasks are scheduled for a minute.
3. Rate limiter will let 600 pollers to go through. 
4. Tasks are scheduled with 100 per second.
5. For 6 seconds polls will return 100 tasks per second.